### ISSUE TITLE
F-Droid build not reproducible

### ISSUE THREAD

**Issue by linsui** [sspanak/tt9#476]
[tt9-v29.0.apk.html.zip](https://github.com/sspanak/tt9/files/14735254/tt9-v29.0.apk.html.zip)

Looks like the command used to get the commit reture different length of string. Could you please take a look? Thanks!

**Comment by sspanak** [comment]
Thanks for reporting!

I have just rebuild the APK from the v29.0 and it is indeed different from the one I uploaded last week. I am not sure why is that, but I'll just try uploading the new one.

Also, one more thing that I've noticed in the logs, is that zipinfo reports different number of files in the archive. Is this to be expected? I would imagine this would result in a different signature. Is there a way to get the entire list of files from the F-droid build, so that I can compare it with my APK?

**Comment by linsui** [comment]
That's expected. I compared an unsiged apk with your signed apk.

**Comment by sspanak** [comment]
So, considering the git hash length, it seems that on my machine it is 8 characters, while on the F-droid build server, git prefers 7 chars. I am not sure why is that. I don't have any other local commits starting with the same 7 characters, so I don't think it's the git "smart" algorithm trying to return the minimum unique hash.

What I can do is: fix the length in the versionName generation function, so it will always be 8 chars long; push one more commit to master, then replace the "v29.0" tag. Is this OK for F-droid? Will be able to handle the same tag but for a new commit?

**Comment by linsui** [comment]
Yes, that works. I'll update the metadata manually anyway.

**Comment by sspanak** [comment]
Alright, let's do it. 

I am a bit busy with personal stuff at the moment, so I can't do much work or testing, but I'll try to do it in the following days and let you know when the new code is pushed.

**Comment by linsui** [comment]
Thanks! Just ping me when it's ready. :)

**Comment by linsui** [comment]
It's fixed, thanks!

**Comment by licaon-kter** [comment]
lets reopen, ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/7632248660

actual diff:
```
--- tt9-v37.0
+++ io.github.sspanak.tt9_643
│   --- tt9-v37.0/assets
├── +++ io.github.sspanak.tt9_643/assets
│ │   --- tt9-v37.0/assets/help.html
│ ├── +++ io.github.sspanak.tt9_643/assets/help.html
│ │ @@ -1 +1 @@
│ │ ├── html2text {}
│ │ │ @@ -1,8 +1,8 @@
│ │ │ +****** Traditional T9 v37.0 ******
│ │ │ -****** Traditional T9 v36.30 ******
│ │ │  This manual explains how to configure and use Traditional T9 in different
│ │ │  scenarios. For installation instructions and information about the "lite" and
│ │ │  "full" versions, please consult the Installation_Guide on GitHub. Finally, you
│ │ │  may want to check out the main_repository_page, which includes all source code,
│ │ │  a developer's guide, the privacy policy, and supplementary documentation.
│ │ │  **** Contents ****
│ │ │     1. Initial_Setup
```

**Comment by sspanak** [comment]
Epic fail! I can't fix this version because it is already on Google Play. I can't modify the release. I can only release the fix in v38.0. Would that be OK?

**Comment by licaon-kter** [comment]
@sspanak yup :)

**Comment by sspanak** [comment]
@licaon-kter, update. It turned out it was more of a stale build cache issue on my side. I have rebuilt the APKs and re-uploaded them to the release. The only difference should be the incorrect version string on the help screen, which is not a big deal for the Google release.

Let's try one more time and if it doesn't work, I'll try to push the new version as soon as possible.

**Comment by licaon-kter** [comment]
reenabled, thanks https://gitlab.com/fdroid/fdroiddata/-/commit/65b260a24a353ac5627cc05f752c22b96f734fa7

**Comment by linsui** [comment]
```
Only in /tmp/tmp8xmxf9t4/tmp_binaries_io.github.sspanak.tt9_774.binary/content/assets: help.html
```

I guess you need to clean up the cache?

**Comment by sspanak** [comment]
@linsui , thanks for reporting. I have uploaded a new APK without the outdated file. If you need to restart the task manually, please, go ahead.

> I guess you need to clean up the cache?

I keep forgetting this all the time. I moved the `.html` files to a different directory but forgot to clean up the project.

**Comment by linsui** [comment]
Thanks!

**Comment by licaon-kter** [comment]
v44 has a Tag but no APK?

**Comment by sspanak** [comment]
@licaon-kter, yes, I haven't made the release publicly visible yet. I would like to test it for 24 more hours, just in case. Do I need to notify you or do something else after publishing, or F-droid will pick it up automatically?

**Comment by linsui** [comment]
Just ping us here. :)

**Comment by licaon-kter** [comment]
@sspanak a git tag is always public, the "pre-release" is a Github thing :)

**Comment by sspanak** [comment]
@licaon-kter , @linsui , v44.0 is now published. 

It wasn't a pre-release, by the way. I kept is as a draft, so it was visible only to me. Either way, it should be OK now.

**Comment by licaon-kter** [comment]
since fdroid checkupdate job picked it up surely it was visible to us all🤷

**Comment by linsui** [comment]
https://gitlab.com/fdroid/checkupdates-bot-fdroiddata/-/jobs/9148165525 45.0 reproducible build failed. It seems caused by https://f-droid.org/en/docs/Reproducible_Builds/#pngs-generated-from-vector-drawables. Could you please take a look? Thanks!

**Comment by sspanak** [comment]
@linsui this is because of the new launcher vector icon. From what I see, the PNGs are used only on API < 21. But since this version, the minimum supported version is 21, so it should be safe to disable them, correct?

**Comment by linsui** [comment]
I thought so. :)

**Comment by sspanak** [comment]
I have updated the release. Let's see how it goes now.

**Comment by linsui** [comment]
Thanks!

**Comment by linsui** [comment]
https://gitlab.com/fdroid/checkupdates-bot-fdroiddata/-/jobs/9281731149 Looks like definitions.yml is not sorted. Could you please take a look? Thanks!

**Comment by sspanak** [comment]
Thanks for reporting! I have uploaded new APKs with the definitions sorted alphabetically. This should ensure identical build results.

**Comment by linsui** [comment]
Thanks!
